Lib/gftools/builder/_ninja.py

```diff
@@ -0,0 +1,360 @@
+"""Ninja file writer for orchestrating font builds"""
+from ninja.ninja_syntax import Writer
+import ninja
+import glyphsLib
+from glyphsLib.builder.builders import UFOBuilder
+import sys
+import ufoLib2
+import yaml
+import os
+from gftools.builder import GFBuilder
+from fontTools.designspaceLib import (
+    DesignSpaceDocument,
+    SourceDescriptor,
+    InstanceDescriptor,
+)
+from pathlib import Path
+
+UNSUPPORTED = ["stylespaceFile", "statFormat4", "ttfaUseScript", "vttSources"]
+
+
+class NinjaBuilder(GFBuilder):
+    def __init__(self, **kwargs):
+        self.old_cwd = os.getcwd()
+        super().__init__(**kwargs)
+
+    def build(self):
+        # In some cases we want to fall back to GFBuilder
+        for unsupported_key in UNSUPPORTED:
+            if self.config.get(unsupported_key):
+                self.logger.error(
+                    "%s configuration parameter not supported by ninja builder, "
+                    "falling back to classic GFBuilder" % unsupported_key
+                )
+                os.chdir(self.old_cwd)
+                raise NotImplementedError()
+
+        self.w = Writer(open("build.ninja", "w"))
+        self.temporaries = []
+        self.setup_rules()
+        self.get_designspaces()
+
+        if self.config["buildVariable"]:
+            self.build_variable()
+            # transfer vf vtt hints now in case static fonts are instantiated
+            if "vttSources" in self.config:
+                self.build_vtt(self.config["vfDir"])
+        if self.config["buildStatic"]:
+            self.build_static()
+            if "vttSources" in self.config:
+                self.build_vtt(self.config["ttDir"])
+        self.w.close()
+
+        ninja_args = []
+        if self.config["logLevel"] == "DEBUG":
+            ninja_args = ["-v", "-j", "1"]
+
+        errcode = ninja._program("ninja", ninja_args)
+
+        # Tidy up stamp files
+        for temporary in self.temporaries:
+            if os.path.exists(temporary):
+                os.remove(temporary)
+
+        if errcode:
+            sys.exit(errcode)
+
+    def setup_rules(self):
+        self.w.comment("Rules")
+        if self.config["logLevel"] == "DEBUG":
+            args = {"pool": "console"}
+        else:
+            args = {}
+        self.w.newline()
+        self.w.comment("Convert glyphs file to UFO")
+        self.w.rule(
+            "glyphs2ufo", "fontmake -o ufo --instance-dir instance_ufo -g $in", **args
```

Comment:
you may want to try passing `--ufo-structure=json`, both when building master UFOs and when building instance UFOs below, once the new fontmake with ufoLib2[json] support gets released (https://github.com/googlefonts/fontmake/pull/1017). It should be a bit faster than writing out the real UFOs -- at least when using `orjson` instead of stdlib json.

Reply:
sure, np. 

> I'm just trying to get this landed...

what's blocking?

Reply:
At this point, I don't think anything is. Hooray!